### PR TITLE
check settings_pack::max_out_request_queue before alert outstanding_r…

### PR DESCRIPTION
**Pull request by tkittich**
…equest_limit_reached

This should fix performance warning: max outstanding piece requests reached, outstanding_request_limit_reached

### CODE REVIEW

**Comment by arvidn** [comment]
The existing check compares against `m_max_out_request_queue` which is set to this setting, [here](https://github.com/arvidn/libtorrent/blob/RC_2_0/src/peer_connection.cpp#L141):
```
m_max_out_request_queue(aux::clamp_assign<std::uint16_t(m_settings.get_int(settings_pack::max_out_request_queue)))
```

This limit can be adjusted if the peer indicates a `reqq` limit in the extended handshake, or if the peer is snubbed (or if the peer is identified as bitcomet).

So, I think you're right to check against the setting instead. I think the existing check can be removed however.

File: src/peer_connection.cpp [hunk 1]
```diff
@@ -5104,6 +5104,7 @@ namespace libtorrent {
 		update_desired_queue_size();
 
 		if (m_desired_queue_size == m_max_out_request_queue
+			&& m_desired_queue_size >= m_settings.get_int(settings_pack::max_out_request_queue)
```
Comment (arvidn) [inline src/peer_connection.cpp:5107]:
```suggestion
		if (m_desired_queue_size >= m_settings.get_int(settings_pack::max_out_request_queue)
```

**Comment by arvidn** [comment]
If you take my suggestion, please add a note to the `Changelog` as well, that this issue was addressed.

**Comment by tkittich** [comment]
> If you take my suggestion, please add a note to the `Changelog` as well, that this issue was addressed.

Thank you for the review & suggestions.

**Review by arvidn — APPROVED** [review]

**Pull request merged**